sdcflows/interfaces/fmap.py

```diff
@@ -663,28 +663,19 @@ def au2rads(in_file, newpath=None):
     """Convert the input phase difference map in arbitrary units (a.u.) to rads."""
     from scipy.stats import mode
     im = nb.load(in_file)
-    data = im.get_fdata(dtype='float32')
+    data = im.get_fdata(caching='unchanged')  # Read as float64 for safety
     hdr = im.header.copy()
 
-    # First center data around 0.0.
-    data -= mode(data, axis=None)[0][0]
+    # Rescale to [0, 2*pi]
+    data = (data - data.min()) * (2 * np.pi / (data.max() - data.min()))
 
-    # Scale lower tail
-    data[data < 0] = - np.pi * data[data < 0] / data[data < 0].min()
-
-    # Scale upper tail
-    data[data > 0] = np.pi * data[data > 0] / data[data > 0].max()
-
-    # Offset to 0 - 2pi
-    data += np.pi
-
-    # Clip
-    data = np.clip(data, 0.0, 2 * np.pi)
+    # Round to float32 and clip
+    data = np.clip(np.float32(data), 0.0, 2 * np.pi)
 
     hdr.set_data_dtype(np.float32)
     hdr.set_xyzt_units('mm')
     out_file = fname_presuffix(in_file, suffix='_rads', newpath=newpath)
-    nb.Nifti1Image(data, im.affine, hdr).to_filename(out_file)
+    nb.Nifti1Image(data, None, hdr).to_filename(out_file)
```

Comment:
If you're keeping the affines in the header, there's no need to pass an affine in to override.